src/priority-queue.ts

```diff
@@ -0,0 +1,262 @@
+import {ArmorCollection} from './collection';
+import {ArmorCollectionSelector} from './selector';
+
+/**
+ * decide what to do when no comparator is passed
+ */
+
+export interface ArmorPriorityQueueOptions<T> {
+	elements?: Array<T>;
+	comparator?: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
+}
+
+const defaultComparator = function (nodeOneIndex: number, nodeTwoIndex: number) {
+	if (typeof nodeOneIndex !== 'number') {
+		return false;
+	}
+	if (typeof nodeTwoIndex !== 'number') {
+		return false;
+	}
+
+	return JSON.stringify(this.elements[nodeOneIndex]) < JSON.stringify(this.elements[nodeTwoIndex]);
+};
+
+export class ArmorPriorityQueue<T> implements ArmorCollection<T> {
+	public elements: T[];
+	public comparator: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
+
+	constructor(options: ArmorPriorityQueueOptions<T> = {}) {
```

Comment:
- [ ] Change comparator type check into a validation of comparator type and throw when it fails (shown below).

Good use of backup default to fallback when comparator is not provided. Due to the complexities of generic types we can't reliably provide a default. Therefore, a user provided comparator is required and we can't do our job without one and it makes sense to throw if no comparator is provided. 

This:
```typescript
		if (typeof options.comparator === 'function') {
			this.comparator = options.comparator;
		} else {
			this.comparator = defaultComparator;
		}
```'
can be condensed down to a throw:
```typescript
if (typeof options.comparator !== 'function') {
   throw new Error('comparator error message here');
}
```